Plugins/MatrixPlugin/MatrixLogic.cs

```diff
@@ -0,0 +1,334 @@
+                GetTotalSkill(CurrentAction.ActionSkill) +
+                CurrentAction.ActionModifier +
+                ActionModifier;
+        
+
+        public int DefenceDicePool => GetTotalSkill(CurrentAction.DefenceSkill) +
+                GetTotalAttribute(CurrentAction.DefenceAttribute) +
+                CurrentAction.DefenceModifier +
+                ActionModifier;
+        
+        public int ActionModifier
+        {
+            get => actionModifier;
+            set
+            {
+                actionModifier = value;
+                OnPropertyChanged();
+                OnPropertyChanged("ActionDicePool");
+                OnPropertyChanged("DefenceDicePool");
+            }
+        }
+
+        public int GetTotalMatrixAttribute(string attribute)
+        {
+            switch (attribute)
+            {
+                case "Attack":
+                    return this.TotalAttack;
+                case "Sleaze":
+                    return this.TotalSleaze;
+                case "Data Processing":
+                    return this.TotalDataProcessing;
+                case "Firewall":
+                    return this.TotalFirewall;
+            }
+            return 0;
+        }
+
+        public int GetTotalSkill(string skill)
+        {
+            if (_character.SkillsSection.GetActiveSkill(skill) != null)
+                return _character.SkillsSection.GetActiveSkill(skill).TotalBaseRating;
+            return 0;
+        }
+        public int GetTotalAttribute(string attribute)
+        {
+            if (_character.GetAttribute(attribute) != null)
+                return _character.GetAttribute(attribute).TotalValue;
+            return 0;
+        }
+
+        private int Parse(string value)
+        {
+            int.TryParse(value, out int result);
+            if (result == 0)
+            {
+                CharacterAttrib attribute = _character.GetAttribute(value.Replace("{", "").Replace("}", ""));
+                if (attribute != null)
+                    result = attribute.Base;
+            }
+            return result;
+        }
+
+        public MatrixLogic(Character character, List<MatrixAction> matrixActions)
+        {
+            this._character = character;
+            Actions = matrixActions;
+            Persons = new List<IHasMatrixAttributes>();
+            Software = new List<Gear>();
+            //Load all CyberDecks,Commlinks and Programs to the Lists
+            foreach (Gear gear in character.Gear)
+                if (gear.Category == "Cyberdecks" || gear.Category == "Commlinks")
+                    Persons.Add(gear);
+                else if (gear.Category.Contains("Program"))
+                    Software.Add(gear);
+                else if (gear.Children.Count > 0)
+                    Software.AddRange(from Gear child in gear.Children
+                                      where gear.Category.Contains("Program")
+                                      select gear);
```

Comment:
You just need to embrace them. Since I use linq regularly this code is totally readable I think. ;)

That being said, this code will break if gear.category is null in any case of the collection.

Reply:
Remove LINQ in https://github.com/chummer5a/chummer5a/pull/4563/commits/cd67c543cb0c14c2a3d4068113bf15cca275a6f9